piecrust/src/store/metadata.rs

```diff
@@ -0,0 +1,56 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::io;
+use std::path::Path;
+use std::sync::Arc;
+
+use rkyv::{archived_root, Deserialize, Infallible};
+
+use crate::module::ModuleMetadata;
+use crate::store::mmap::Mmap;
+
+/// Module metadata pertaining to a given module but maintained by the host.
+#[derive(Debug, Clone)]
+pub struct Metadata {
+    mmap: Arc<Mmap>,
+    data: ModuleMetadata,
+}
```

Comment:
It doesn't seem that we need to keep the `mmap` field here. Once it is deserialized into memory we can discard the mmap, no?

Reply:
And if this is the case, this struct is unnecessary, since the `from_file` function could be on `ModuleMetadata` instead.

Reply:
Problem is that we are inserting Metadata to ModuleData, so we need to have `from_file` which produces Metadata. With `mmap` elimination - I don't see how to do it without extra serializations. Issue is that Metadata is needed in serialized form as it is being stored along with ModuleData, so if we eliminate `mmap`, we end up serializing on demand which will be wasteful.

Reply:
I think we can tackle this in #181